subprojects/core/src/main/java/org/gradle/api/internal/tasks/DefaultTaskContainer.java

```diff
@@ -198,12 +211,6 @@ public DynamicObject getTasksAsDynamicObject() {
 
     public void realize() {
         project.getModelRegistry().realizeNode(modelNode.getPath());
-
-        new CachingDirectedGraphWalker<Task, Void>(new DirectedGraph<Task, Void>() {
```

Comment:
I thought some more about this and the only use case is when there is a task dependency on a task which will be created by a rule. These will no longer be shown. But I think that's acceptable, because we show rules at the end of the report.